src/bebone/core/types/vector/vector2_type.tpp

```diff
@@ -91,88 +92,113 @@ namespace bebone::core {
     const Vec2<float> Vec2<T>::down{ 0.0f, -1.0f };
 
     template<typename T>
-    Vec2<T> Vec2<T>::operator+(const Vec2& vec) const { return Vec2(x + vec.x, y + vec.y); }
+    Vec2<T> Vec2<T>::operator +(const Vec2& v) const { return Vec2(x + v.x, y + v.y); }
 
     template<typename T>
-    Vec2<T> Vec2<T>::operator-(const Vec2& vec) const { return Vec2(x - vec.x, y - vec.y); }
+    Vec2<T> Vec2<T>::operator -(const Vec2& v) const { return Vec2(x - v.x, y - v.y); }
 
     template<typename T>
-    Vec2<T> Vec2<T>::operator*(const Vec2& vec) const { return Vec2(x * vec.x, y * vec.y); }
+    Vec2<T> Vec2<T>::operator *(const Vec2& v) const { return Vec2(x * v.x, y * v.y); }
 
     template<typename T>
-    Vec2<T> Vec2<T>::operator/(const Vec2& vec) const { return Vec2(x / vec.x, y / vec.y); }
+    Vec2<T> Vec2<T>::operator /(const Vec2& v) const { return Vec2(x / v.x, y / v.y); }
 
     template<typename T>
-    Vec2<T> Vec2<T>::operator+(const T& value) const { return Vec2(x + value, y + value); }
-    
-    template<typename T>
-    Vec2<T> Vec2<T>::operator-(const T& value) const { return Vec2(x - value, y - value); }
+    Vec2<T>& Vec2<T>::operator +=(const Vec2<T>& v) { x += v.x; y += v.y; return *this; }
 
     template<typename T>
-    Vec2<T> Vec2<T>::operator*(const T& value) const { return Vec2(x * value, y * value); }
-    
+    Vec2<T>& Vec2<T>::operator -=(const Vec2<T>& v) { x -= v.x; y -= v.y; return *this; }
+
     template<typename T>
-    Vec2<T> Vec2<T>::operator/(const T& value) const { return Vec2(x / value, y / value); }
+    Vec2<T>& Vec2<T>::operator *=(const Vec2<T>& v) { x *= v.x; y *= v.y; return *this; }
 
-    /* Vector x= Vector */
     template<typename T>
-    Vec2<T>& Vec2<T>::operator+=(const Vec2<T>& other) { this->x += other.x; this->y += other.y; return *this; }
+    Vec2<T>& Vec2<T>::operator /=(const Vec2<T>& v) { x /= v.x; y /= v.y; return *this; }
 
     template<typename T>
-    Vec2<T>& Vec2<T>::operator-=(const Vec2<T>& other) { this->x -= other.x; this->y -= other.y; return *this; }
+    Vec2<T> Vec2<T>::operator +(const T& scalar) const { return Vec2(x + scalar, y + scalar); }
     
     template<typename T>
-    Vec2<T>& Vec2<T>::operator*=(const Vec2<T>& other) { this->x *= other.x; this->y *= other.y; return *this; }
+    Vec2<T> Vec2<T>::operator -(const T& scalar) const { return Vec2(x - scalar, y - scalar); }
+
+    template<typename T>
+    Vec2<T> Vec2<T>::operator *(const T& scalar) const { return Vec2(x * scalar, y * scalar); }
     
     template<typename T>
-    Vec2<T>& Vec2<T>::operator/=(const Vec2<T>& other) { this->x /= other.x; this->y /= other.y; return *this; }
+    Vec2<T> Vec2<T>::operator /(T scalar) const {
+        scalar = 1.0f / scalar;
+        return *this * scalar;
```

Comment:
bro you cooking 😎

Reply:
![21702d26d7b34450795c4d5ac22c3ef1](https://github.com/user-attachments/assets/07090d95-3f06-4073-92eb-132622861a12)